test/end-to-end/offer.js

```diff
@@ -633,41 +632,139 @@ describe("Offer endpoint tests", () => {
+                title: "Secret Offer",
+                publishDate: "2021-01-14T00:00:00.000Z",
+                publishEndDate: "2021-02-28T00:00:00.000Z",
+                description: "Secret offer that only admins and the owner can see",
+                contacts: ["geral@niaefeup.pt", "229417766"],
+                jobType: "SUMMER INTERNSHIP",
+                fields: ["FRONT_END", "BACK_END"],
+                technologies: ["Flutter", "Node.js"],
+                isHidden: true,
+                // owner: Will be set in beforeAll,
+                location: "Area 51, Secret State, 321",
+            };
+
+            let test_1, test_2, test_3;
+
+            const test_agent = agent();
+
+            const test_user_admin = {
+                email: "admin@email.com",
+                password: "password123",
+            };
+            const test_user_company = {
+                email: "company@email.com",
+                password: "password123",
+            };
+
             beforeAll(async () => {
-                await Offer.create([test_offer_1, test_offer_2]);
+                await Company.deleteMany({});
+                await Offer.deleteMany({});
+                await Account.deleteMany({});
+
+                test_company = await Company.create({
+                    name: "test company",
+                    bio: "a bio",
+                    contacts: ["a contact"]
+                });
+
+                await Account.create({
+                    email: test_user_admin.email,
+                    password: await hash(test_user_admin.password),
+                    isAdmin: true
+                });
+                await Account.create({
+                    email: test_user_company.email,
+                    password: await hash(test_user_company.password),
+                    company: test_company._id
+                });
+                [test_1, test_2, test_3] =
+                    await Promise.all([test_offer_1, test_offer_2, test_offer_3].map(async (offer) => {
+                        offer.owner = test_company._id.toString();
+                        const { _id } = await Offer.create(offer);
+                        offer["_id"] = _id.toString();
+                        return offer;
+                    }));
+                // test_user_non_admin = {
+                //     email: "company@email.com",
+                //     password: "password123",
+                // };
+            });
+
+            test("should return offer", async () => {
+                let res = await test_agent.get(`/offers/${test_1["_id"]}`);
+                expect(res.status).toBe(HTTPStatus.OK);
+                const extracted_data_1 = res.body;
+
+                res = await test_agent.get(`/offers/${test_2["_id"]}`);
+                expect(res.status).toBe(HTTPStatus.OK);
+                const extracted_data_2 = res.body;
+
+                expect(extracted_data_1).toMatchObject(test_1);
```

Comment:
Move this above (before the second request) so that if it fails, no unnecessary requests are made